Zend/zend_compile.c

```diff
@@ -1767,6 +1769,10 @@ static bool zend_try_ct_eval_class_const(zval *zv, zend_string *class_name, zend
 		return 0;
 	}
 
+	if (ZEND_CLASS_CONST_FLAGS(cc) & ZEND_ACC_DEPRECATED) {
```

Comment:
I think this is redundant, given that `zend_verify_ct_const_access` already has this check.

Reply:
Indeed, I was just trying to get this to work